gnomad/utils/annotations.py

```diff
@@ -1795,3 +1796,65 @@ def _agg_by_group(
     )
 
     return ht.drop("cols")
+
+
+def update_structured_annotations(
+    ht: hl.Table,
+    annotation_update_exprs: Dict[str, hl.Expression],
+    annotation_update_label: Optional[str] = None,
+) -> hl.Table:
+    """
+    Update highly structured annotations on a Table.
+
+    This function recursively updates annotations defined by `annotation_update_exprs`
+    and if `annotation_update_label` is supplied, it checks if the sample annotations
+    are different from the input and adds a flag to the Table, indicating which
+    annotations have been updated for each sample.
+
+    :param ht: Input Table with structured annotations to update.
+    :param annotation_update_exprs: Dictionary of annotations to update, structured as
+        they are structured on the input `ht`.
+    :param annotation_update_label: Optional string of the label to use for an
+        annotation indicating which annotations have been updated. Default is None, so
+        no annotation is added.
+    :return: Table with updated annotations and optionally a flag indicating which
+        annotations were changed.
+    """
+
+    def _update_struct(
+        struct_expr: hl.expr.StructExpression,
+        update_exprs: Union[Dict[str, hl.Expression], hl.expr.Expression],
+    ) -> Tuple[Dict[str, Any], hl.expr]:
```

Comment:
is the first one not always `Dict[str, hl.expr.BooleanExpression]`? and why `hl.expr`?

Reply:
I had this warning for the return in else: 
```
Expected type 'tuple[dict[str, BooleanExpression], StructExpression]', got 'tuple[dict[str, Any], Expression]' instead: 1846. 
```

Reply:
Try this `Tuple[Dict[str, hl.expr.BooleanExpression], Any]` I think the first should be a Boolean, the suppose the second can be multiple different things